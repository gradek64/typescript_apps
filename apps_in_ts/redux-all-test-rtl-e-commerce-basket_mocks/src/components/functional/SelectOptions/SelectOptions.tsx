import React from 'react';

interface IProps {
  options: (number | string)[];
  eventHandler: (e: React.ChangeEvent) => void;
  defaultV?: number | string;
}

const SelectOptions = ({ options = [1, 2, 3, 4, 5], eventHandler, defaultV }: IProps) =>
(
  <select onChange={eventHandler} value={defaultV}>
    { options.map((option, i) => <option key={`option${i}`} value={option}>{option}</option>)}
  </select>
);
export default SelectOptions;
