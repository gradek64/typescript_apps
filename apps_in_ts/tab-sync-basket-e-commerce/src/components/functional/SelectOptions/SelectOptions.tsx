import React from 'react';

interface IProps {
  options: (number | string)[];
  eventHandler: (e: React.ChangeEvent) => void;
  defaultV?: number | string;
}

const SelectOptions = ({ options, eventHandler, defaultV }: IProps) => {

  console.log('defaultV', defaultV)

  return (
    <select onChange={eventHandler}>
      { options.map((option, i) => <option key={`option${i}`} value={option} selected={defaultV === option}>{option}</option>)}
    </select>
  );
}
export default SelectOptions;
