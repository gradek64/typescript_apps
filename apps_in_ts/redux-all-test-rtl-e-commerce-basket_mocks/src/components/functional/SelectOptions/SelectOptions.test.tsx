import React from 'react';
import { render, fireEvent } from '@testing-library/react'

import SelectOptions from './SelectOptions';
const props = {
  options: [1, 2, 3, 4, 5],
  eventHandler: jest.fn(),
  defaultV: 1
}
describe('render', () => {
  it('renders correctly with default value', () => {
    const { asFragment } = render(<SelectOptions {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
describe('events', () => {
  it('executes event handler on change', () => {
    const { eventHandler } = props
    const { container } = render(<SelectOptions {...props} />);
    const select = container.querySelector('select')
    fireEvent.change(select as HTMLSelectElement, { target: { value: '2' } })
    expect(eventHandler).toHaveBeenCalled()
  })
})
