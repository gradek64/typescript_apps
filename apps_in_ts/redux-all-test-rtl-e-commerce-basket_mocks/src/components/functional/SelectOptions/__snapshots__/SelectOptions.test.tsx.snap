// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly with default value 1`] = `
<DocumentFragment>
  <select>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
  </select>
</DocumentFragment>
`;
